core/src/main/java/org/mule/runtime/core/policy/DefaultPolicyManager.java

```diff
@@ -38,27 +40,33 @@
 
   private Collection<OperationPolicyParametersTransformer> operationPolicyParametersTransformerCollection = emptyList();
   private Collection<SourcePolicyParametersTransformer> sourcePolicyParametersTransformerCollection = emptyList();
+  private Collection<SourcePolicyPointcutParametersFactory> sourcePointcutFactories = emptyList();
```

Comment:
PolicyManager does not look cohesive. There are two sets of methods that are used in different contexts, one for sources, and another one for operations. 
The relation between the findXPolicyInstance and the createXPointcutParameters is strange too, as they will be always used together, so why not merge both methods in one?